src/engine/Service.cpp

```diff
@@ -92,6 +95,44 @@ ResultTable Service::computeResult() {
   serviceIriString.remove_suffix(1);
   ad_utility::httpUtils::Url serviceUrl{serviceIriString};
 
+  if (siblingTree_ != nullptr) {
+    // Get the result of the siblingTree, to (potentially)
+    // reduce complexity of the SERVICE query.
+    auto siblingResult = siblingTree_->getResult();
+
+    const size_t rowLimit = 100;
+    if (siblingResult->size() < rowLimit) {
```

Comment:
If you do it the other way round (if size >= limit => immediately return/break), then the code gets less nested.